### PR TITLE
Create config for iFlight BLITZ_H7_PRO

### PR DESCRIPTION
**Pull request by KarateBrot**
Please be aware of `ADCX_DMA_OP` and `TIMUPX_DMA_OPT`. I'm not sure this is set up correctly.

@haslinghuis @SteveCEvans @blckmn Can you give me a hint where to look at (to confirm the correct setup)?

### CODE REVIEW

File: configs/IFLIGHT_BLITZ_H7_PRO/config.h [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+#define ADC1_DMA_OPT        8
+#define ADC3_DMA_OPT        9
+#define TIMUP1_DMA_OPT      0
+#define TIMUP2_DMA_OPT      0
+#define TIMUP3_DMA_OPT      2
+#define TIMUP4_DMA_OPT      1
+#define TIMUP5_DMA_OPT      0
+#define TIMUP8_DMA_OPT      0
+
+#define DEFAULT_GYRO_TO_USE             GYRO_CONFIG_USE_GYRO_1
```
Comment (haslinghuis) [inline configs/IFLIGHT_BLITZ_H7_PRO/config.h:126]: GYRO_1 is used by default.

```suggestion
```

**Comment by KarateBrot** [comment]
> If `dma show` shows that the ADC/timers are assigned channels then you’ve nothing to do. Thanks to the very flexible DMA assignment on the H7 this should be OK.

Thanks, then this part is correctly set up. I followed the setup of a reference FC from iFlight.

File: configs/IFLIGHT_BLITZ_H7_PRO/config.h [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    TIMER_PIN_MAP(2,  PA0,  2,  2) \
+    TIMER_PIN_MAP(3,  PA1,  2,  3) \
+    TIMER_PIN_MAP(4,  PA2,  2,  4) \
+    TIMER_PIN_MAP(5,  PA3,  2,  5) \
+    TIMER_PIN_MAP(6,  PD12, 1,  6) \
+    TIMER_PIN_MAP(7,  PD13, 1,  7) \
+    TIMER_PIN_MAP(8,  PD14, 1, 12) \
+    TIMER_PIN_MAP(9,  PD15, 1, -1) \
+    TIMER_PIN_MAP(10, PE5,  1,  0) \
+    TIMER_PIN_MAP(11, PE6,  1, -1) \
```
[the remark this answers is not in the document]
Comment (KarateBrot) [reply configs/IFLIGHT_BLITZ_H7_PRO/config.h:109]: This is the timer pin mapping Betaflight returned to me after typing `timer map` into the CLI. Can I safely remove the undefined pin entries?

[the remark this answers is not in the document]
Comment (haslinghuis) [reply configs/IFLIGHT_BLITZ_H7_PRO/config.h:109]: Until now always did `TIMER_PIN_MAPPING` entries manually.

As the pins are not in the resource they should be save to remove.
What is the output of `resource show all` ?

[the remark this answers is not in the document]
Comment (KarateBrot) [reply configs/IFLIGHT_BLITZ_H7_PRO/config.h:109]:
```
# resource show all
Currently active IO resource assignments:
(reboot to update)
--------------------
A00: MOTOR 3
A01: MOTOR 4
A02: FREE
A03: FREE
A04: FREE
A05: SPI_SCK 1
A06: SPI_SDI 1
A07: FREE
A08: FREE
A09: FREE
A10: FREE
A11: USB
A12: USB
A13: SWD
A14: SWD
A15: BEEPER
B00: MOTOR 1
B01: MOTOR 2
B02: GYRO_EXTI
B03: SPI_SCK 3
B04: SPI_SDI 3
B05: SPI_SDO 3
B06: I2C_SCL 1
B07: I2C_SDA 1
B08: FREE
B09: FREE
B10: I2C_SCL 2
B11: I2C_SDA 2
B12: OSD_CS
B13: SPI_SCK 2
B14: SPI_SDI 2
B15: SPI_SDO 2
C00: ADC_BATT
C01: ADC_CURR
C02: FREE
C03: FREE
C04: ADC_EXT
C05: FREE
C06: FREE
C07: FREE
C08: SDIO_D0
C09: SDIO_D1
C10: SDIO_D2
C11: SDIO_D3
C12: SDIO_CK
C13: FREE
C14: FREE
C15: GYRO_CS 1
D00: FREE
D01: FREE
D02: SDIO_CMD
D03: FREE
D04: FREE
D05: FREE
D06: FREE
D07: SPI_SDO 1
D08: FREE
D09: FREE
D10: PINIO 1
D11: PINIO 2
D12: FREE
D13: FREE
D14: FREE
D15: FREE
E00: FREE
E01: FREE
E02: FREE
E03: LED 2
E04: LED 3
E05: FREE
E06: FREE
E07: FREE
E08: FREE
E09: FREE
E10: FREE
E11: FREE
E12: SPI_SCK 4
E13: SPI_SDI 4
E14: SPI_SDO 4
E15: FREE
F00: FREE
F01: FREE
F02: FREE
F03: FREE
F04: FREE
F05: FREE
F06: FREE
F07: FREE
F08: FREE
F09: FREE
F10: FREE
F11: FREE
F12: FREE
F13: FREE
F14: FREE
F15: FREE
G00: FREE
G01: FREE
G02: FREE
G03: FREE
G04: FREE
G05: FREE
G06: FREE
G07: FREE
G08: FREE
G09: FREE
G10: FREE
G11: FREE
G12: FREE
G13: FREE
G14: FREE
G15: FREE

Currently active Timers:
-----------------------
TIM1: FREE
TIM2: FREE
TIM3: FREE
TIM4: FREE
TIM5: FREE
TIM6: FREE
TIM7: FREE
TIM8:
    CH1 : DSHOT_BITBANG 2
    CH2 : DSHOT_BITBANG 1
TIM12: FREE
TIM13: FREE
TIM14: FREE
TIM15: FREE
TIM16: FREE
TIM17: FREE

Currently active DMA:
--------------------
DMA1 Stream 0: DSHOT_BITBANG 2
DMA1 Stream 1: DSHOT_BITBANG 1
DMA1 Stream 2: SPI_SDO 1
DMA1 Stream 3: SPI_SDI 1
DMA1 Stream 4: SPI_SDO 2
DMA1 Stream 5: SPI_SDI 2
DMA1 Stream 6: FREE
DMA1 Stream 7: FREE
DMA2 Stream 0: ADC 1
DMA2 Stream 1: ADC 3
DMA2 Stream 2: FREE
DMA2 Stream 3: FREE
DMA2 Stream 4: FREE
DMA2 Stream 5: FREE
DMA2 Stream 6: FREE
DMA2 Stream 7: FREE
```

[the remark this answers is not in the document]
Comment (haslinghuis) [reply configs/IFLIGHT_BLITZ_H7_PRO/config.h:109]: Would just remove them. Perhaps there is bug in the `timer map` which needs investigation outside of this PR.

**Comment by SteveCEvans** [comment]
What does `timer` rather than `timer show` give you? Make sure everything that uses timers is enabled. So enable `LED_STRIP`, set `dshot_bitbang=off` etc.

**Comment by KarateBrot** [comment]
> What does `timer` rather than `timer show` give you? Make sure everything that uses timers is enabled. So enable `LED_STRIP`, set `dshot_bitbang=off` etc.

I don't know what else uses timers but I did what you said and it looks good to me.

## Output of `timer`
```
timer B00 AF2
# pin B00: TIM3 CH3 (AF2)
timer B01 AF2
# pin B01: TIM3 CH4 (AF2)
timer A00 AF2
# pin A00: TIM5 CH1 (AF2)
timer A01 AF2
# pin A01: TIM5 CH2 (AF2)
timer A02 AF2
# pin A02: TIM5 CH3 (AF2)
timer A03 AF2
# pin A03: TIM5 CH4 (AF2)
timer D12 AF2
# pin D12: TIM4 CH1 (AF2)
timer D13 AF2
# pin D13: TIM4 CH2 (AF2)
timer D14 AF2
# pin D14: TIM4 CH3 (AF2)
timer D15 AF2
# pin D15: TIM4 CH4 (AF2)
timer E05 AF4
# pin E05: TIM15 CH1 (AF4)
timer E06 AF4
# pin E06: TIM15 CH2 (AF4)
timer A08 AF1
# pin A08: TIM1 CH1 (AF1)
timer A15 AF1
# pin A15: TIM2 CH1 (AF1)
timer C07 AF3
# pin C07: TIM8 CH2 (AF3)
timer C06 AF3
# pin C06: TIM8 CH1 (AF3)
timer B08 AF1
# pin B08: TIM16 CH1 (AF1)
timer B09 AF1
# pin B09: TIM17 CH1 (AF1)
```

## Output of `timer show`
```
Currently active Timers:
-----------------------
TIM1:
    CH1 : LED_STRIP
TIM2: FREE
TIM3:
    CH3 : MOTOR 1
    CH4 : MOTOR 2
TIM4: FREE
TIM5:
    CH1 : MOTOR 3
    CH2 : MOTOR 4
TIM6: FREE
TIM7: FREE
TIM8: FREE
TIM12: FREE
TIM13: FREE
TIM14: FREE
TIM15: FREE
TIM16: FREE
TIM17: FREE
```

## Output of `dma show`
```
Currently active DMA:
--------------------
DMA1 Stream 0: MOTOR 1
DMA1 Stream 1: MOTOR 2
DMA1 Stream 2: MOTOR 3
DMA1 Stream 3: MOTOR 4
DMA1 Stream 4: SPI_SDO 1
DMA1 Stream 5: SPI_SDI 1
DMA1 Stream 6: SPI_SDO 2
DMA1 Stream 7: SPI_SDI 2
DMA2 Stream 0: ADC 1
DMA2 Stream 1: ADC 3
DMA2 Stream 2: FREE
DMA2 Stream 3: FREE
DMA2 Stream 4: FREE
DMA2 Stream 5: FREE
DMA2 Stream 6: LED_STRIP
DMA2 Stream 7: FREE
```

**Comment by haslinghuis** [comment]
> PD14, PD15, PE5 and PE6 are not defined.

PD14 is S9
PD15 is S10
PE5 is S11
PE6 is S12

Are these servo pins ?

**Comment by KarateBrot** [comment]
Apparently they are servo pins, as verified by iFlight. I am pushing a new commit with the changes.

**Review by haslinghuis — APPROVED** [review]

**Review by sugaarK — APPROVED** [review]

**Pull request merged**